library/Core/src/Juvix/Core/Translate.hs

```diff
@@ -95,6 +99,10 @@ irToHRWith ::
   HR.Term primTy primVal
 irToHRWith pats t = fst $ exec pats (varsTerm t) $ irToHR' t
 
+-- | @irToHR'@ transforms an IR term into an HR one. works like
+-- @hrToIR'@ but in reverse. Namely we have binders introduce a new
+-- name to the stack, then the free var index, @n@, grabs the @nth@
+-- item in the stack
```

Comment:
I think I'd prefer the same style as I suggested in comments above -- explain the dual purposes of `irToHR'` and `irElimToHR'`, where the former generates names (using `withFresh`) while the latter looks them up (`getPatToSym`; `lookupIndex`).